src/orders/service/order-details.service.ts

```diff
@@ -0,0 +1,46 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { OrderDetailsRepository } from '../repository/order-details.repository';
+import { OrderDetail } from '../domain/order-detail.domain';
+import { ProductsRepository } from 'src/products/repository/products.repository';
+import { LocationsRepository } from 'src/products/repository/locations.repository';
+
+@Injectable()
+export class OrderDetailsService {
+  constructor(
+    private readonly orderDetailsRepository: OrderDetailsRepository,
+    private readonly producstRepository: ProductsRepository,
+    private readonly locationsRepository: LocationsRepository,
+  ) {}
+
+  create(orderDetail: OrderDetail) {
+    const products = this.producstRepository.findAll();
+    const location = this.locationsRepository.findOne(orderDetail.location.id);
+    const order = this.orderDetailsRepository.findOne(orderDetail.order.id);
+    if (!products) {
```

Comment:
Cool that you added some validations here! However, stock update is missing. 

This service should not call the repositories from other modules. It should have dependencies to the services instead.